pkg/tfvars/vsphere/vsphere.go

```diff
@@ -76,7 +104,99 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 		OvaFilePath:       cachedImage,
 		PreexistingFolder: sources.PreexistingFolder,
 		DiskType:          sources.DiskType,
+
+		NetworkZone:            sources.NetworkZone,
+		FolderZone:             folderZone,
+		VCenters:               vcenterZones,
+		DeploymentZone:         deploymentZones,
+		FailureDomainZone:      failureDomainZones,
+		ControlPlaneConfigZone: controlPlaneConfigZone,
 	}
 
 	return json.MarshalIndent(cfg, "", "  ")
 }
+
+func createFolderZoneMap(infraID string, deploymentZone map[string]*vtypes.DeploymentZoneSpec, failureDomainZones map[string]vtypes.FailureDomainSpec) map[string]*folder {
+	folders := make(map[string]*folder)
+
+	for k, v := range deploymentZone {
+		tempFolder := new(folder)
+
+		tempFolder.Datacenter = failureDomainZones[v.FailureDomain].Topology.Datacenter
+		tempFolder.Name = v.PlacementConstraint.Folder
+
+		if tempFolder.Name == "" {
+			tempFolder.Name = infraID
+			deploymentZone[k].PlacementConstraint.Folder = infraID
```

Comment:
This does not seem like the right place to be setting this value. Should probably be in the machine pool defaults.

Reply:
We are using `Folder` directly in terraform which is why it needs to be defaulted:
https://github.com/openshift/installer/pull/6064/files#diff-d0b75770b7d6a46f012401b853e416a64c1dc1abbfdc5867698a93ebf098cbbbR75

Its optional for a user but not for terraform, is machinepool defaults where that should be? 
More Zones could be defined than used.

Reply:
> Its optional for a user but not for terraform, is machinepool defaults where that should be?

This is what I was thinking of... it seems to be consistent with what you are saying
https://github.com/openshift/installer/blob/master/pkg/types/vsphere/machinepool.go#L42-L65

Reply:
> This is what I was thinking of... it seems to be consistent with what you are saying

is it?

I need to set the a platform-spec defined struct parameter to the InfraID, which the machinepool doesn't include.
And is infraID available when that runs?